src/services/AbstractPalletsService.ts

```diff
@@ -41,15 +43,18 @@ export abstract class AbstractPalletsService extends AbstractService {
 	):
 		| Option<StorageMetadataV13>
 		| Option<PalletStorageMetadataV14>
-		| Option<PalletErrorMetadataV14> {
+		| Option<PalletErrorMetadataV14>
```

Comment:
I Think this function can be as simple as:
```typescript
	private getPalletMetadataType(
		meta: ModuleMetadataV13 | PalletMetadataV14,
		metadataFieldType: IMetadataFieldType
	): PalletMetadata {
			return this.getProperty(meta as PalletMetadataV14, metadataFieldType);
	}
```

We would just need to add the following types:

```typescript
type PalletMetadata = | Option<StorageMetadataV13>
	| Option<PalletStorageMetadataV14>
	| Option<PalletErrorMetadataV14>
	| Option<PalletEventMetadataV14>

type IMetadataFieldType = 'events' | 'storage' | 'errors';
```

Then in `findPalletMeta` set `metadataFieldType` to `IMetadataFieldType`.

Reply:
awesome refactor! Ill update this